torch/_higher_order_ops/out_dtype.py

```diff
@@ -123,15 +123,10 @@ def out_dtype_autograd(
     output_dtype: torch.dtype,
     *args
 ):
-    # TODO: maybe support autograd
-    flat_operands, _ = pytree.tree_flatten(args)
-    if torch.is_grad_enabled() and any(
-        f.requires_grad for f in flat_operands if isinstance(f, torch.Tensor)
-    ):
-        raise RuntimeError("Autograd is not supported for out_dtype")
-
     with torch._C._AutoDispatchBelowAutograd():
-        return out_dtype(op, output_dtype, *args)
+        result = out_dtype(op, output_dtype, *args)
+        # TODO: maybe support autograd
+        return autograd_not_implemented(result, args, "out_dtype", delayed=False)
```

Comment:
Hmm, `autograd_not_implemented` should probably just accept an operator and return a function.

```
out_dtype.py_impl(DispatchKey.Autograd)(autograd_not_implemented(out_dtype))
```
This way authors of HigherOrderOperator don't need to remember they need a `with torch._C._AutoDispatchBelowAutograd():`

Reply:
Also, in the future (if you're up to it, or someone else can do this), we should just have the default autograd behavior on HigherOrderOp be this autograd_not_implemented thing so that users don't need to remember to add it themselves.